src/net/sourceforge/kolmafia/session/AutumnatonManager.java

```diff
@@ -87,6 +91,37 @@ private static void parseQuest(final String responseText, final int snarfblat) {
         "autumnatonQuestTurn", KoLCharacter.getTurnsPlayed() + calculateQuestTurns(questNumber));
   }
 
+  private static final Pattern SNARFBLAT_PATTERN = Pattern.compile("heythereprogrammer=(\\d+)");
+
+  public static int getSnarfblat(final String urlString) {
+    Matcher matcher = SNARFBLAT_PATTERN.matcher(urlString);
+    return matcher.find() ? StringUtilities.parseInt(matcher.group(1)) : null;
+  }
+
+  public static boolean registerRequest(final String urlString) {
+    int decision = ChoiceUtilities.extractOptionFromURL(urlString);
+    switch (decision) {
+      case 1 -> {
+        // Installing an upgrade
+        int choice = ChoiceUtilities.extractChoiceFromURL(urlString);
+        String desc = ChoiceManager.choiceDescription(choice, decision);
+        RequestLogger.printLine(desc);
+        RequestLogger.updateSessionLog(desc);
+        return true;
+      }
+      case 2 -> {
+        // Sending on a mission
+        int snarfblat = getSnarfblat(urlString);
```

Comment:
I suppose this can be null

Reply:
Unexpected, but, I suppose.
getSnarfblat returns an int - but if not found, it returns null? And Java accepted that?
And looking up KoLAdventure by snarfblat can return null if unknown adventure.

Both fixed.